### PR TITLE
Behavior of dependency check has changed

### PR DESCRIPTION
**Pull request by ronny-mysten**
Previously, a flag needed to be added to have `sui client publish` command verify dependencies. That is now the default behavior with a new flag, `--skip-dependency-verification`, available to bypass that behavior.

### CODE REVIEW

File: doc/src/build/cli-client.md [hunk 1]
```diff
@@ -631,9 +632,7 @@ Supply a package path to `verify-source` (or run from package root) to have the
 
 `sui client verify-source ./code/MyPackage`
 
-The default behavior is for the command to verify only the direct source of the package, but you can supply the `--verify-deps` flag to have the command verify dependencies, as well. If you just want to verify dependencies, you can also add the `--skip-source` flag. Attempting to use the `--skip-source` flag without including the `--verify-deps` flag results in an error because there is essentially nothing to verify.  
-
-Running `sui client verify-source --skip-source --verify-deps` does not publish the package, but is otherwise the same as `sui client publish --verify-dependencies`, as described in the [previous section](#publish-packages).
```
Comment (amnn) [inline doc/src/build/cli-client.md:636]: Might be worth keeping some version of this sentence that explains which invocation of `verify-source` matches the check that publish (now) does by default?  I can imagine people wanting to check that that test will pass without necessarily publishing when it does.

**Review by amnn — APPROVED** [review]
LGTM, thanks @ronny-mysten !  Left a small comment about one of the sections that was removed.

**Comment by randall-Mysten** [comment]
I think this warrants a mention in doc-history please

**Review by randall-Mysten — APPROVED** [review]

File: doc/src/build/cli-client.md (hunk 1 above)
Comment (ronny-mysten) [reply doc/src/build/cli-client.md:636]: Makes sense. Will add.

**Pull request merged**